plugins/libvirt/kvm_io

```diff
@@ -59,54 +63,59 @@ def clean_vm_name(vm_name):
             pass
     return re.sub(r"[^a-zA-Z0-9_]", "_", vm_name)
 
+
 def fetch(vms):
     ''' Fetch values for a list of pids
     @param dictionary {kvm_pid: cleaned vm name}
     '''
     res = {}
     for pid in vms:
-        f = open("/proc/%s/io" % pid, "r")
+        f = open("/proc/%s/io" % pid.decode("utf-8"), "r")
         for line in f.readlines():
             if "read_bytes" in line:
                 read = line.split()[1]
-                print "%s_read.value %s" % (vms[pid], read)
+                print("%s_read.value %s" % (vms[pid], read))
             if "write_bytes" in line:
                 write = line.split()[1]
-                print "%s_write.value %s" % (vms[pid], write)
+                print("%s_write.value %s" % (vms[pid], write))
                 break
         f.close()
 
+
 def detect_kvm():
     ''' Check if kvm is installed
     '''
     kvm = Popen("which kvm", shell=True, stdout=PIPE)
     kvm.communicate()
     return not bool(kvm.returncode)
 
+
 def find_vm_names(pids):
     '''Find and clean vm names from pids
     @return a dictionary of {pids : cleaned vm name}
     '''
     result = {}
     for pid in pids:
-        cmdline = open("/proc/%s/cmdline" % pid, "r")
-        result[pid] = clean_vm_name(re.sub(r"^.*guest=([a-zA-Z0-9.-_-]*).*$",r"\1", cmdline.readline()))
+        cmdline = open("/proc/%s/cmdline" % pid.decode("utf-8"), "r")
+        result[pid] = clean_vm_name(re.sub(r"^.*guest=([a-zA-Z0-9.-_-]*).*$", r"\1", cmdline.readline()))
     return result
 
+
 def list_pids():
     ''' Find the pid of kvm processes
     @return a list of pids from running kvm
     '''
     pid = Popen("pidof qemu-kvm qemu-system-x86_64 kvm", shell=True, stdout=PIPE)
     return pid.communicate()[0].split()
 
+
 if __name__ == "__main__":
     if len(sys.argv) > 1:
         if sys.argv[1] in ['autoconf', 'detect']:
             if detect_kvm():
-                print "yes"
+                print("yes")
             else:
-                print "no"
+                print("no")
```

Comment:
Please add a reason for the failure in brackets (e.g. `no (kvm executable missing)`).